core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -167,6 +167,9 @@ private TableProperties() {
   public static final String ORC_BATCH_SIZE = "read.orc.vectorization.batch-size";
   public static final int ORC_BATCH_SIZE_DEFAULT = 5000;
 
+  public static final String INCLUDE_HIDDEN_PATHS = "read.fs.include-hidden-paths";
+  public static final boolean INCLUDE_HIDDEN_PATHS_DEFAULT = false;
```

Comment:
I don't think that this is a good candidate for a table property. Instead, I think it should be an option on the `DeleteOrphanFiles` action:

```java
SparkActions.get(spark)
    .deleteOrphanFiles(table)
    .option("ignore-hidden-paths", "false")
    .execute();
```

Table properties are used for more generally useful options. But this one is fairly specific to the action itself.